clang/lib/3C/RewriteUtils.cpp

```diff
@@ -598,7 +597,20 @@ void RewriteConsumer::emitRootCauseDiagnostics(ASTContext &Context) {
           // SL is invalid when the File is not in the current translation unit.
           if (SL.isValid()) {
             EmittedDiagnostics.insert(PSL);
-            DE.Report(SL, ID) << PtrCount << WReason.second.getWildPtrReason();
+            DE.Report(SL, ID) << PtrCount << WReason.second.getReason();
+          }
+          for (auto &Note : WReason.second.additionalNotes()) {
+            PersistentSourceLoc NPSL = Note.Location;
+            llvm::ErrorOr<const clang::FileEntry *> NFile =
+                SM.getFileManager().getFile(NPSL.getFileName());
+            if (!NFile.getError()) {
+              SourceLocation NSL = SM.translateFileLineCol(
+                  *File, NPSL.getLineNo(), NPSL.getColSNo());
```

Comment:
```suggestion
                  *NFile, NPSL.getLineNo(), NPSL.getColSNo());
```
Looks to me like a possible mistake.  We could confirm it by doing a test with two files.